framework/src/main/java/org/checkerframework/framework/util/element/SuperTypeApplier.java

```diff
@@ -73,10 +73,10 @@ public int getElementIndex() {
         return index;
     }
 
-    /** @return the type_index of anno's TypeAnnotationPosition */
+    /** @return the pos of anno's TypeAnnotationPosition */
     @Override
     public int getTypeCompoundIndex(Attribute.TypeCompound anno) {
-        return anno.getPosition().type_index;
+        return anno.getPosition().pos;
```

Comment:
I don't think this is the right change. `pos` is the source tree position of the annotation; for an annotation from bytecode this is just -1, the default value.
The result of this method is compared against the result of `getElementIndex()` (in IndexedElementAnnotationApplier).
As `pos` and `index` count completely different things, the values only match in corner cases.

Can you add test cases where `pos` and `index` are not -1 and ensure they work?
For example, what happens when you use an interface instead of `String`?